dynamiqs/utils/utils.py

```diff
@@ -550,30 +551,81 @@ def todm(x: Tensor) -> Tensor:
         )
 
 
-def ket_overlap(x: Tensor, y: Tensor) -> Tensor:
-    r"""Returns the overlap (inner product) $\braket{\varphi,\psi}$ between two kets
-    $\ket\varphi$ and $\ket\psi$.
+def overlap(x: Tensor, y: Tensor) -> Tensor:
+    r"""Returns the overlap between two quantum states.
+
+    For pure quantum states $\ket\psi$ and $\ket\varphi$, the overlap is defined by
+    $\lvert\braket{\varphi|\psi}\rvert^2$. For density matrices $\rho$ and $\sigma$,
+    the overlap is defined by $\tr{\sigma^\dagger \rho}$.
```

Comment:
Not sure about this one. I'm leaning a bit more towards keeping the `dag` because it seems more mathematically precise, but agree it's more annoying to document.

Reply:
Let's keep the `\dag`, I agree.